### PR TITLE
Referral migrations

### PR DESCRIPTION
**Pull request by megaherz**

### CODE REVIEW

File: internal/storage/storage.go [hunk 1]
```diff
@@ -18,12 +18,37 @@ var ErrAddressIsTaken = fmt.Errorf("address is taken")
...
+// Referral ...
+type Referral struct {
+	Owner          string         `db:"sender"`
+	Receiver       string         `db:"receiver"`
+	Status         ReferralStatus `db:"status"`
+	RegisteredAt   time.Time      `db:"registered_at"`
+	InstalledAt    sql.NullTime   `db:"installed_at"`
+	ConfirmedAt    sql.NullTime   `db:"confirmed_at"`
+	SenderReward   sql.NullInt32  `db:"sender_reward"`
+	ReceiverReward sql.NullInt32  `db:"receiver_reward"`
```
Comment (mikluke) [inline internal/storage/storage.go:51]: It's better to use sdk.Dec

File: scripts/migrations/postgres/20211109185301_referral.up.sql [hunk 2]
```diff
@@ -0,0 +1,88 @@
+CREATE EXTENSION IF NOT EXISTS "pgcrypto";
```
Comment (mikluke) [inline scripts/migrations/postgres/20211109185301_referral.up.sql:1]: I think we can just use hash of something. It's simpler than brute-forcing unique value. or use SERIAL + some salt

File: internal/storage/storage.go (hunk 1 above)
Comment (megaherz) [reply internal/storage/storage.go:51]: Does it support Scan/Value form the sql package?

File: scripts/migrations/postgres/20211109185301_referral.up.sql (hunk 2 above)
Comment (megaherz) [reply scripts/migrations/postgres/20211109185301_referral.up.sql:1]: The trigger generates a unique 8 letters code with a good selectivity. There is going to be only one loop in most cases.

File: internal/storage/storage.go (hunk 1 above)
Comment (mikluke) [reply internal/storage/storage.go:54]: not sure, but I believe it does

File: scripts/migrations/postgres/20211109185301_referral.up.sql (hunk 2 above)
Comment (mikluke) [reply scripts/migrations/postgres/20211109185301_referral.up.sql:1]: okay, shouldn't we have a ttl index for that column? when a referral's it doesn't make any sense

File: internal/storage/postgres/postgres.go [hunk 3]
```diff
@@ -71,11 +69,31 @@ func (p pg) UpsertRequest(ctx context.Context, owner, email, address, code strin
...
 			return storage.ErrAddressIsTaken
 		}
 		return fmt.Errorf("failed to exec query: %w", err)
 	}
 
 	return nil
 }
+
+func (p pg) CreateReferral(ctx context.Context, sender, receiver string) error {
+	if _, err := p.db.ExecContext(ctx, `INSERT INTO referral VALUES ($1, $2, $3, CURRENT_TIMESTAMP, NULL, NULL, NULL)`,
```
Comment (mikluke) [inline internal/storage/postgres/postgres.go:82]: NamedExec would be better

File: internal/storage/storage.go (hunk 1 above)
Comment (megaherz) [reply internal/storage/storage.go:54]: checked, it does not.

File: internal/storage/postgres/postgres.go (hunk 3 above)
Comment (megaherz) [reply internal/storage/postgres/postgres.go]: Done

File: scripts/migrations/postgres/20211109185301_referral.up.sql (hunk 2 above)
Comment (megaherz) [reply scripts/migrations/postgres/20211109185301_referral.up.sql:1]: What is the purpose of TTL? The referral code is not going to be expired, but the whole referral program might.

**Pull request merged**